File: BitManipulation/136_SingleNumber.cpp
/*
 * @Author: xuezaigds@gmail.com
 * @Last Modified time: 2016-04-11 23:02:26
 */

class Solution {
public:
    int singleNumber(vector<int>& nums) {
        int xor_n = 0;
        for(const auto &n: nums){
            xor_n ^= n;
        }
        return xor_n;
    }
};

File: BitManipulation/136_SingleNumber.py
#! /usr/bin/env python
# -*- coding: utf-8 -*-


class Solution(object):
    def singleNumber(self, nums):
        num = nums[0]
        for i in nums[1:]:
            num = num ^ i
        return num

"""
[1]
[1,2,3,4,4,3,2]
"""
